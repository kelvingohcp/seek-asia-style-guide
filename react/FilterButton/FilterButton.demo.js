import { FilterButton } from 'seek-asia-style-guide/react';

export default {
  route: '/filterbutton',
  title: 'FilterButton',
  component: FilterButton,
  initialProps: {
    children: 'Hello world',
    color: 'primary',
    isActive: false
  },
  options: [
    {
      label: 'Colour',
      type: 'radio',
      states: [
        {
          label: 'Primary',
          transformProps: props => ({
            ...props,
            color: 'primary'
          })
        },
        {
          label: 'Secondary',
          transformProps: props => ({
            ...props,
            color: 'secondary'
          })
        }
      ]
    }
  ]
};
