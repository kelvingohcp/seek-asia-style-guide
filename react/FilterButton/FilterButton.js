import styles from './FilterButton.less';
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Button from '../Button/Button';

const FilterButton = props => {
  const { className, color, isActive, ...restProps } = props;
  const isPrimary = color === 'primary';
  const isSecondary = color === 'secondary';
  return (
    <Button
      {...restProps}
      color={color}
      className={classnames(className, {
        [styles.root]: isPrimary || isSecondary,
        [styles.root_primary]: isPrimary,
        [styles.root_secondary]: isSecondary,
        [styles.isActive]: isActive
      })}
    />
  );
};

FilterButton.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  isActive: PropTypes.bool
};

export default FilterButton;
