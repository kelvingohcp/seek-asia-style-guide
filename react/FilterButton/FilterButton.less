@import (reference) "~seek-asia-style-guide/theme";

.root {
  &.isActive,
  &:focus,
  &:active {
    outline: none;
    box-shadow:
      0 0 4px 2px @activeGlow,
      inset 0 0 0 1px @activeGlow;
  }

  &.ishover,
  &:hover {
    box-shadow: inset 0 0 0 1px @activeGlow;
  }
}

.root_secondary {
  &:hover {
    background-color: @actionPastelBlue;
  }
}

.root_primary {
  &:hover {
    background-color: @actionBlue;
  }
}
