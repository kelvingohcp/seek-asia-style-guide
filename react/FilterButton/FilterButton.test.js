import React from 'react';
import { shallow } from 'enzyme';
import FilterButton from './FilterButton';

describe('FilterButton:', () => {
  describe('color variants:', () => {
    it('should render primary FilterButton', () => {
      const wrapper = shallow(<FilterButton color="primary">SEEK</FilterButton>);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render secondary FilterButton', () => {
      const wrapper = shallow(<FilterButton color="secondary">SEEK</FilterButton>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should render with node', () => {
    const wrapper = shallow(
      <FilterButton color="primary">
        <h5>SEEK</h5>
      </FilterButton>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with array of nodes', () => {
    const wrapper = shallow(
      <FilterButton color="primary">
        <span>SEEK</span>
        <span>AU</span>
      </FilterButton>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with className', () => {
    const wrapper = shallow(
      <FilterButton color="primary" className="foo">
        SEEK
      </FilterButton>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with active state', () => {
    const wrapper = shallow(
      <FilterButton color="primary" isActive className="foo">
        SEEK
      </FilterButton>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
