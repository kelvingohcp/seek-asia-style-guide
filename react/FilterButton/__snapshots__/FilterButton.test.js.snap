// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton: color variants: should render primary FilterButton 1`] = `
<Button
  className="root root_primary"
  color="primary"
  component="button"
>
  SEEK
</Button>
`;

exports[`FilterButton: color variants: should render secondary FilterButton 1`] = `
<Button
  className="root root_secondary"
  color="secondary"
  component="button"
>
  SEEK
</Button>
`;

exports[`FilterButton: should render with active state 1`] = `
<Button
  className="foo root root_primary isActive"
  color="primary"
  component="button"
>
  SEEK
</Button>
`;

exports[`FilterButton: should render with array of nodes 1`] = `
<Button
  className="root root_primary"
  color="primary"
  component="button"
>
  <span>
    SEEK
  </span>
  <span>
    AU
  </span>
</Button>
`;

exports[`FilterButton: should render with className 1`] = `
<Button
  className="foo root root_primary"
  color="primary"
  component="button"
>
  SEEK
</Button>
`;

exports[`FilterButton: should render with node 1`] = `
<Button
  className="root root_primary"
  color="primary"
  component="button"
>
  <h5>
    SEEK
  </h5>
</Button>
`;
